test/x509_internal_test.c

```diff
@@ -221,10 +223,326 @@ static int tests_X509_PURPOSE(void)
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = tests[i].NotAfter + 1;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+
+        test_time = 1442939232;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = 1443004020;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+
+        test_time = MIN_UTC_TIME;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = MIN_UTC_TIME - 1;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+
+        test_time = MAX_UTC_TIME;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+
+        test_time = MAX_UTC_TIME + 1;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+
+        /* Test integer value boundaries */
+        test_time = INT64_MIN;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = INT32_MIN;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = -1;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = 0;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = 1;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = INT32_MAX;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = UINT32_MAX;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit, upper_limit);
+        test_time = INT64_MAX;
+        failures += test_a_time(ctx, x509, test_time, tests[i].NotBefore,
+                                tests[i].NotAfter, lower_limit,
```

Comment:
I like the change. I just would like the test output to be more useful in case something will eventually fail. My concern is we will get just breif message something has failed in a test run and then it will become quite hard to find which of the tests here failed. thanks.